core/src/main/java/org/stellar/anchor/event/EventService.java

```diff
@@ -1,14 +1,94 @@
 package org.stellar.anchor.event;
 
+import java.util.List;
 import org.stellar.anchor.api.event.AnchorEvent;
+import org.stellar.anchor.api.exception.AnchorException;
 import org.stellar.anchor.api.exception.EventPublishException;
-import org.stellar.anchor.sep24.Sep24Transaction;
-import org.stellar.anchor.sep31.Sep31Transaction;
 
+/**
+ * The EventService is used to publish events to the event queue and to read events from the event
+ * queue.
+ *
+ * <p>Read event example
+ *
+ * <pre>
+ *     EventService eventService = new EventServiceImpl();
+ *     eventService.publish(new TransactionEvent(...));
+ *     eventService.publish(new ControlEvent(...));
+ *     ...
+ *     EventService.Session session = eventService.createSession(EventService.EventQueue.TRANSACTION);
+ *     EventService.ReadResponse readResponse = session.read();
+ *     List&lt;AnchorEvent&gt; events = readResponse.getEvents();
+ *     ...
+ *     session.ack(readResponse);
+ *     ...
+ *     session.close();
+ *     ...
+ *     eventService = null;
+ * </pre>
+ *
+ * <p>Publish event example
+ *
+ * <pre>
+ *     EventService eventService = new EventServiceImpl();
+ *     EventService.Session session = eventService.createSession(EventService.EventQueue.TRANSACTION);
+ *     session.publish(new TransactionEvent(...));
+ *     session.close();
+ * </pre>
+ */
 public interface EventService {
-  void publish(Sep24Transaction txn, AnchorEvent.Type type);
+  /**
+   * To be deprecated. Use createSession() instead.
+   *
+   * @param event the event to be published
+   * @throws EventPublishException
+   */
+  void publish(AnchorEvent event) throws EventPublishException;
 
-  void publish(Sep31Transaction txn, AnchorEvent.Type type);
+  /**
+   * Creates a session for publishing and reading events.
+   *
+   * @return a session object.
+   */
+  Session createSession(EventQueue eventQueue);
 
-  void publish(AnchorEvent event) throws EventPublishException;
+  interface Session {
+    /**
+     * Publishes an event to the event queue. The queue will be determined by the implementation of
+     * the Session.
+     *
+     * @param event the event to publish
+     */
+    void publish(AnchorEvent event) throws AnchorException;
+
+    /**
+     * Reads events from the event queue.
+     *
+     * @return
+     */
+    ReadResponse read() throws AnchorException;
+
+    /**
+     * Acknowledges that the events returned by the read() method have been processed.
+     *
+     * @param readResponse
+     */
+    void ack(ReadResponse readResponse) throws AnchorException;
+
+    /** Closes the session. */
+    void close() throws AnchorException;
+  }
+
+  interface ReadResponse {
+    List<AnchorEvent> getEvents();
+  }
+
+  /** List of events queues that are supported by the EventService. */
+  enum EventQueue {
+    /** The event queue for events that are to be processed by the anchor. */
+    TRANSACTION,
```

Comment:
This is for SEP-24/31 transactions where as control is for callback?

Reply:
No. The `CONTROL` is the place-holder for the [control-plane ](https://www.snaplogic.com/blog/data-plane-vs-control-plane-whats-the-difference#:~:text=The%20control%20plane%20decides%20how,instructions%20by%20forwarding%20the%20packets.). The comment is updated to

```java
  enum EventQueue {
    /** The event queue for the transaction events that are to be processed by the anchor. */
    TRANSACTION,

    /** The event queue for events that are used for control-plane purposes. */
    CONTROL
  }
```

Reply:
Are we using a queue for control plane related messages today?

Reply:
No, it is not for related message today. It's just a hidden placeholder.